File: Domain/Fretter.Domain/Entities/Fretter/ContratoTransportadorArquivoTipo.cs
using Fretter.Domain.Enum;
using System;
using System.Collections.Generic;

namespace Fretter.Domain.Entities
{
    public class ContratoTransportadorArquivoTipo : EntityBase
    {
        #region "Construtores"
        protected ContratoTransportadorArquivoTipo() { }
        public ContratoTransportadorArquivoTipo(int Id, int? EmpresaId, int TransportadorId, int ImportacaoArquivoTipoItemId, string Alias)
        {
            this.Ativar();
            this.Id = Id;
            this.EmpresaId = EmpresaId;
            this.TransportadorId = TransportadorId;
            this.ImportacaoArquivoTipoItemId = ImportacaoArquivoTipoItemId;
            this.Alias = Alias;
        }

        #endregion

        #region "Propriedades"
        public int? EmpresaId { get; protected set; }
        public int TransportadorId { get; protected set; }
        public int ImportacaoArquivoTipoItemId { get; protected set; }
        public string Alias { get; protected set; }
        #endregion

        #region "Referencias"
        public ImportacaoArquivoTipoItem ImportacaoArquivoTipoItem { get; set; }
        #endregion

        #region "Métodos"
        public void AtualizarEmpresaId(int EmpresaId) => this.EmpresaId = EmpresaId;
        public void AtualizarTransportadorId(int TransportadorId) => this.TransportadorId = TransportadorId;
        public void AtualizarImportacaoArquivoTipoItemId(int ImportacaoArquivoTipoItemId) => this.ImportacaoArquivoTipoItemId = ImportacaoArquivoTipoItemId;
        public void AtualizarAlias(string Alias) => this.Alias = Alias;
        #endregion
    }
}

File: Web/Fretter.Web/src/app/core/models/contratoTransportadorArquivoTipo.ts
import { EntityBase } from "./entityBase";
import { ImportacaoArquivoTipoItem } from "./importacaoArquivoTipoItem";

export class ContratoTransportadorArquivoTipo extends EntityBase {

  constructor() {
    super();
    this.id = 0;
    this.importacaoArquivoTipoItemId = 0;
    this.transportadorId = 0;
    this.ativo = true;
    this.importacaoArquivoTipoItem = new ImportacaoArquivoTipoItem();
  }
	
  empresaId	: number;				
  transportadorId	: number;				
  importacaoArquivoTipoItemId : number;		
  alias: string;
  aliasList: Array<string> = new Array<string>();

  importacaoArquivoTipoItem: ImportacaoArquivoTipoItem  = new ImportacaoArquivoTipoItem();

  tipoArquivo : string;
  tipoCobranca : string;
}
